src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -538,6 +548,16 @@ public async Task<ReturnResult> RenameAsync(IStorageItemWithPath source, string
 			var progress = new Progress<FileSystemProgress>();
 			progress.ProgressChanged += (s, e) => returnStatus = returnStatus < ReturnResult.Failed ? e.Status!.Value.ToStatus() : returnStatus;
 
+			if (string.IsNullOrWhiteSpace(newName)
+				|| ContainsRestrictedCharacters(newName)
+				|| ContainsRestrictedFileName(newName))
```

Comment:
This condition is tested twice (at least) in the file. It should be extracted inside a method called `IsSuitedForFileName(string name)` or something akin.